ironic/drivers/modules/drac/raid.py

```diff
@@ -876,13 +968,84 @@ def _check_node_raid_jobs(self, task):
+                        'delete_foreign_config':
+                    controllers = list()
+                    jobs_required = False
+                    for controller_id in \
+                            node.driver_internal_info[
+                                'raid_config_parameters']:
+                        controller_cap = clear_foreign_config(
+                            node, controller_id)
+                        controller = {'raid_controller': controller_id,
+                                      'is_reboot_required':
+                                          controller_cap[
+                                              'is_reboot_required']}
+                        controllers.append(controller)
+                        jobs_required = jobs_required or controller_cap[
+                            'is_commit_required']
+
+                    if not jobs_required:
+                        node.driver_internal_info[
+                            'raid_config_substep'
+                        ] = 'change_disk_state'
+                        LOG.info(
+                            "No foreign drives detected,"
+                            "performing next substep : {}".format(
+                                node.driver_internal_info[
+                                    'raid_config_substep']))
+                    else:
+                        _commit_to_controllers(
+                            node,
+                            controllers,
+                            substep='change_disk_state')
+
+                if node.driver_internal_info['raid_config_substep'] == \
+                        'change_disk_state':
+                    # In change_phy_disks state to jbod after clear reset
+                    # configuration and deleting foreign drives if any
+
+                    client = drac_common.get_drac_client(node)
+                    jbod = drac_common.drac_constants.RaidStatus.jbod
+                    change_disk_state = \
+                        client.change_physical_disk_state(mode=jbod)
+                    controllers = list()
+                    for controller_id in \
+                            node.driver_internal_info[
+                                'raid_config_parameters']:
+                        controller = {'raid_controller': controller_id,
+                                      'is_reboot_required':
+                                          change_disk_state[
+                                              'is_reboot_required']}
+                        controllers.append(controller)
+
+                    _commit_to_controllers(
+                        node,
+                        controllers, substep='completed')
+
+                if node.driver_internal_info['raid_config_substep'] == \
+                        'completed':
+                    self._clear_raid_substep(node)
+                    self._resume_cleaning(task)
```

Comment:
Yes i just tried removing self.resume_cleaning() on this line and it  executed successfully but i  will changed the flow as you changes which seems more  accurate.